cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/TaxPage.java

```diff
@@ -0,0 +1,452 @@
+
+    @FindBy(name = "oiotherInvestmentIncome")
+    private WebElement OIOtherInvestmentIncomeField;
+
+    @FindBy(name = "oinetBusinessIncome")
+    private WebElement OINetBusinessIncomeField;
+
+    @FindBy(name = "oiadditionalIncome")
+    private WebElement OIAdditionalIncomeField;
+
+    //webelements from tax estimator form - Deductions
+
+    @FindBy(id = "deduction-save-button")
+    private WebElement deductSaveButton;
+
+    @FindBy(id = "deduction-edit-button")
+    private WebElement deductEditButton;
+
+    @FindBy(id = "deduction-delete-button")
+    private WebElement deductDeleteButton;
+
+    @FindBy(id = "deduction-add-button")
+    private WebElement deductAddButton;
+
+    @FindBy(id = "deduction-submit-button")
+    private WebElement deductSubmitButton;
+
+    @FindBy(name = "deddeduction")
+    private WebElement deductDeductionTypeField;
+
+    @FindBy(name = "dedamountSpent")
+    private WebElement deductDeductionAmountField;
+
+    //webelements from tax estimator form - Review and submit
+
+
+    @FindBy(id = "dreview-w2-prev-button")
+    private WebElement reviewW2PreviousButton;
+
+    @FindBy(id = "review-w2-next-button")
+    private WebElement reviewW2NextButton;
+
+    @FindBy(id = "review-deduct-prev-button")
+    private WebElement reviewDeductPreviousButton;
+
+    @FindBy(id = "review-deduct-next-button")
+    private WebElement reviewDeductNextButton;
+
+    @FindBy(id = "review-submit-button")
+    private WebElement reviewSubmitButton;
+
+    //Buttons across top of tax page
+
+    @FindBy(id = "file-taxes")
+    private WebElement fileTaxesButton;
+
+    @FindBy(id = "estimate-refund")
+    private WebElement estimateRefundButton;
+
+    @FindBy(id = "document-checklist")
+    private WebElement documentChecklistButton;
+
+    @FindBy(id = "refund-planning")
+    private WebElement refundPlanningButton;
+
+
+
+    //Login Page
+    // private static final String url = "http://localhost:5173/login";
+
+    // public void getMain() {
+    //     try {
+    //         Thread.sleep(1000);
+    //     } catch (InterruptedException e) {
+    //         e.printStackTrace();
+    //     }
+    //     this.driver.get(url);
+    // }
+
+    //login to site
+    // public void login() {
+    //     try {
+    //         Thread.sleep(1000);
+    //     } catch (InterruptedException e) {
+    //         e.printStackTrace();
+    //     }
+    //     emailField.sendKeys("joseph.sam@gmail.com");
+    //     passwordField.sendKeys("password1");
+    //     loginButton.click();
+
+    // }
+    // //Navigate to tax page
+    // public void NavigateToTaxPage() {
+    //     try {
+    //         Thread.sleep(1000);
+    //     } catch (InterruptedException e) {
+    //         e.printStackTrace();
+    //     }
+        
+    //     taxLink.click();
+    // }
+
+    public boolean checkForListOfExistingTaxEstimationRecords() {
+        try {
+            Thread.sleep(1000);
```

Comment:
Low impact: Sleeping the thread for a second can cause the testing suite to take much longer than needed.
Suggestion: Use a wait.until(ExpectedCondition.<method>) or similar.

Reply:
I used this in a few places and now I am scared to change it.

Reply:
I eliminated some of the thread sleeps. Are they not permitted at all?